include/boost/url/detail/storage.hpp

```diff
@@ -58,37 +62,53 @@ class alloc_storage
         return cap_;
     }
 
+    BOOST_URL_NODISCARD
     char*
-    resize(std::size_t n) override
+    reserve(std::size_t n) override
     {
-        if(n <= cap_)
+        if(n > cap_)
```

Comment:
The reason I use two `return` statements is it makes the coverage report better. The way you have it written, there is no way to know that the case where `n <= cap_` is taken.